### PR TITLE
refactor(assert,expect): use shared `format.ts` and `diff.ts`

### LINKED ISSUE #3912: move `_format.ts` into root
**Is your feature request related to a problem? Please describe.**

followup of https://github.com/denoland/deno_std/pull/3886#issuecomment-1835302006. currently `assert/_format.ts` and `expect/_format.ts` is identical but duplicated across modules.

**Describe the solution you'd like**

create private top-level module, `_format.ts` and `_format_test.ts`

**Describe alternatives you've considered**

- symlinks would probably not work on web.
- keep the status quo, however this increases cognitive complexity and makes it harder to sync both files.

### PR DESCRIPTION
**Pull request by iuioiua**
Closes #3912

### CODE REVIEW

File: internal/mod.ts
Comment (kt3k) [inline internal/mod.ts:1]: Is this file necessary?

**Comment by kt3k** [comment]
This seem copying files from `assert` to `internal`, but test cases seem having better names in `expect`. How about copying these utils from `expect/` instead?

File: internal/mod.ts
Comment (iuioiua) [reply internal/mod.ts:1]: Yes. Check out the file diff. Some files import from both `internal/format.ts` and `internal/diff.ts`.  For such cases, we can use `mod.ts` for now.

**Review by kt3k — APPROVED** [review]
LGTM
